libs/@local/graph/authorization/src/policies/components.rs

```diff
@@ -0,0 +1,189 @@
+use alloc::borrow::Cow;
+use std::collections::HashSet;
+
+use error_stack::{Report, ResultExt as _};
+use type_system::{
+    knowledge::entity::id::EntityEditionId, ontology::VersionedUrl, principal::actor::ActorId,
+};
+
+use super::{
+    Context, ContextBuilder, PolicySet,
+    principal::actor::AuthenticatedActor,
+    store::{PolicyStore, PrincipalStore, error::ContextCreationError},
+};
+
+#[derive(Debug)]
+pub struct PolicyComponents {
+    pub actor_id: Option<ActorId>,
+    pub policy_set: PolicySet,
+    pub context: Context,
+}
+
+impl PolicyComponents {
+    #[must_use]
+    pub fn builder<S>(store: &S) -> PolicyComponentsBuilder<'_, S> {
+        PolicyComponentsBuilder::new(store)
+    }
+}
+
+pub struct PolicyComponentsBuilder<'a, S> {
```

Comment:
Would it maybe make sense to use something like https://docs.rs/bon/latest/bon/ here instead?